### PR TITLE
new resources and localization helper

### PR DESCRIPTION
**Pull request by mgreenegit**
This change is [Reviewable](https://reviewable.io/reviews/powershell/xactivedirectory/81)

### CODE REVIEW

File: DSCResources/MSFT_xADComputerProperty/MSFT_xADComputerProperty.schema.mof [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[ClassVersion("1.0.0.0"), FriendlyName("xADComputerProperty")]
+class MSFT_xADComputerProperty : OMI_BaseResource
+{
+    [Key, Description("The name of the computer object in Active Directory.")] String Name;
+    [Key, Description("The name of the property of the computer object in Active Directory."), ValueMap{"OperatingSystem","OperatingSystemVersion"}, Values{"OperatingSystem","OperatingSystemVersion"}] String Property;
+    [Required, Description("The value of the property of the computer object in Active Directory.")] String Value;
+};
```
Comment (TravisEz13) [inline DSCResources/MSFT_xADComputerProperty/MSFT_xADComputerProperty.schema.mof]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: DSCResources/MSFT_xADServiceAccountOnComputer/MSFT_xADServiceAccountOnComputer.schema.mof [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[ClassVersion("1.0.0.0"), FriendlyName("xADServiceAccountOnComputer")]
+class MSFT_xADServiceAccountOnComputer : OMI_BaseResource
+{
+    [Write, Description("An enumerated value that describes if the Active Directory service account is installed on the machine.\nPresent {default}  \nAbsent   \n"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
+    [Key, Description("Specifies an Active Directory service account object by account name.")] String Identity;
+};
```
Comment (TravisEz13) [inline DSCResources/MSFT_xADServiceAccountOnComputer/MSFT_xADServiceAccountOnComputer.schema.mof]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: DSCResources/MSFT_xSPN/MSFT_xSPN.Schema.mof [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[ClassVersion("1.0.0.0"), FriendlyName("xSPN")]
+class MSFT_xSPN : OMI_BaseResource
+{
+    [Write, Description("An enumerated value that describes if the SPN is installed.\nPresent {default}  \nAbsent   \n"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
+    [Key, Description("The full SPN name including SPN Type EG: HOST/Server.")] String SPN;
+    [Key, Description("The name of the Computer or User the SPN is associated to.")] String ObjectName;
+    [Write, Description("An enumerated value that describes if the ObjectName is of type 'Server' or 'User'.\nServer {default}  \nUser   \n"), ValueMap{"Server","User"}, Values{"Server","User"}] String ObjectType;
+};
```
Comment (TravisEz13) [inline DSCResources/MSFT_xSPN/MSFT_xSPN.Schema.mof]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: xActiveDirectoryHelper.psm1 [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+            Write-Verbose "Loading $ModuleName Module"
+           
+            $CurrentVerbose = $VerbosePreference
+            $VerbosePreference = "SilentlyContinue"
+            $null = Import-Module -Name $ModuleName -ErrorAction Stop
+            $VerbosePreference = $CurrentVerbose
+        }
+    }
+}
```
Comment (TravisEz13) [inline xActiveDirectoryHelper.psm1]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

**Comment by TravisEz13** [comment]
It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)  Please work with us to come up with a plan to cover this code.

Suggestion: Base the tests on the [Unit tests Template](https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1)

File: xActiveDirectoryHelper.psm1 (hunk 4 above)
Comment (iainbrighton) [inline xActiveDirectoryHelper.psm1]: There's already a helper module: https://github.com/PowerShell/xActiveDirectory/blob/dev/DSCResources/MSFT_xADCommon/MSFT_xADCommon.ps1. Surely, these functions should be added here?

File: xActiveDirectory.psd1 [hunk 5]
```diff
@@ -1,6 +1,6 @@
 @{
 # Version number of this module.
-ModuleVersion = '2.9.0.0'
```
Comment (TravisEz13) [inline xActiveDirectory.psd1]: Please do not change the module or class version in resources per the [contribution guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#creating-a-new-dsc-resource-in-an-existing-module)

**Comment by TravisEz13** [comment]
Reviewed 1 of 8 files at r1, 9 of 9 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[xActiveDirectory.psd1, line 3 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KED1H4dEONGqq_iLDLB:-KED1H4dEONGqq_iLDLC:653137616) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/xActiveDirectory.psd1#L3)):*
Please do not change the module or class version in a resources per the [contribution guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#creating-a-new-dsc-resource-in-an-existing-module)

---

*[xActiveDirectoryHelper.psm1, line 99 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KE2vp0mYK3EFyoQH-9--r2-99:-KED1yGp9ArAR4wheuln:-1702652214) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/xActiveDirectoryHelper.psm1#L99)):*
I agree that they should be consolidated, but this could be done after the merge as an issue.

---

*[DSCResources/MSFT_xADComputerProperty/MSFT_xADComputerProperty.psm1, line 118 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KED1_Mznh6TyFw6WRmB:-KED1_Mznh6TyFw6WRmC:2127475188) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/DSCResources/MSFT_xADComputerProperty/MSFT_xADComputerProperty.psm1#L118)):*
Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

---

*[DSCResources/MSFT_xADServiceAccountOnComputer/MSFT_xADServiceAccountOnComputer.psm1, line 124 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KED1eDAXKOsdzDl31B1:-KED1eDAXKOsdzDl31B2:2127475188) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/DSCResources/MSFT_xADServiceAccountOnComputer/MSFT_xADServiceAccountOnComputer.psm1#L124)):*
Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

---

*[DSCResources/MSFT_xSPN/MSFT_xSPN.psm1, line 209 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KED1glWsl4YeUSavcSx:-KED1glWsl4YeUSavcSy:2127475188) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/DSCResources/MSFT_xSPN/MSFT_xSPN.psm1#L209)):*
Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/powershell/xactivedirectory/81)*

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.  Please consider closing and re-opening when ready for review.

---

Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[xActiveDirectory.psd1, line 2 \[r2\]](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-KFNKgsXIfZF4UngGtB_:-KFNKgsXIfZF4UngGtBa:2049161513) ([raw file](https://github.com/powershell/xactivedirectory/blob/7dc7352a343647da1ba02c60054454d30eba07ba/xActiveDirectory.psd1#L2)):*
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xactivedirectory/81#-:-KFNKp5rgXuC1x2_3wx7:2053084077)*

**Comment by TravisEz13** [comment]
Closing because the pull request has not been updated or feedback given for 21 days.  Once you update the code, please feel free to reopen the pull request.  Thanks, Travis.

**Pull request closed without merging**